### PR TITLE
Connect related products components to Redux store

### PR DESCRIPTION
**Pull request by VincentCYChen**
- updated sample store as well

### CODE REVIEW

File: client/src/redux/reducers/relatedProductReducer.js [hunk 1]
```diff
@@ -1,4 +1,45 @@
-const defaultState = [];
+const defaultState = [
```
Comment (chengmuy) [inline client/src/redux/reducers/relatedProductReducer.js]: Is this just for development? If so, let's make sure we create a Trello card to remove it later on

File: client/src/redux/actions/related.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
-const setRelatedProduct = (product) => ({
-  type: 'SET_RELATED_PRODUCT',
+export const SET_RELATED_PRODUCT = 'SET_RELATED_PRODUCT';
```
Comment (chengmuy) [inline client/src/redux/actions/related.js]: Why is this being exported?

File: client/src/components/related-items/RelateAndCompare.js [hunk 3]
```diff
@@ -12,14 +13,23 @@ class RelateAndCompare extends React.Component {
...
 
-export default RelateAndCompare;
+const mapStateToProps = (state) => {
+  return {
+    relatedProducts: state.relatedProducts
+  };
+};
+
+export default connect(mapStateToProps, {
+  setRelatedProduct
```
Comment (chengmuy) [inline client/src/components/related-items/RelateAndCompare.js]: Does this work? I don't see `dispatch` being mapped anywhere?

File: client/src/redux/reducers/relatedProductReducer.js (hunk 1 above)
Comment (VincentCYChen) [reply client/src/redux/reducers/relatedProductReducer.js]: Yes! just added the card.

File: client/src/components/related-items/RelateAndCompare.js (hunk 3 above)
Comment (VincentCYChen) [reply client/src/components/related-items/RelateAndCompare.js]: I think connect uses dispatch under the hood to dispatch actions.

File: client/src/redux/actions/related.js (hunk 2 above)
Comment (VincentCYChen) [reply client/src/redux/actions/related.js]: Because if I set it to a variable VS code will help auto fill instead of typing out all the "type".

Comment (VincentCYChen) [reply client/src/redux/actions/related.js]: When I am writing code in my components

Comment (chengmuy) [reply client/src/redux/actions/related.js]: not sure I follow - you don't actually refer to "SET_RELATED_PRODUCT" elsewhere in your code do you?

Comment (VincentCYChen) [reply client/src/redux/actions/related.js]: Not yet, but I will use it in my components when I refer to my reducer later.

File: client/src/components/related-items/RelateAndCompare.js (hunk 3 above)
Comment (chengmuy) [reply client/src/components/related-items/RelateAndCompare.js]: cool, didn't know you could use the API like this!

**Pull request merged**